docs/specs/README.md

```diff
@@ -0,0 +1,62 @@
+# Guidelines for Feature Request(s)
+
+This document covers, how to contribute new feature(s) or significant design changes to BigchainDB. Since, contribution
+of such functionality involves design discussion and might take multiple releases to complete,
+we expect the contributors to follow certain guidelines before proceeding with the implementation.
+
+
+## Classification
+How to classify your contribution as a feature and determine if it needs design discussion. 
+
+Here are some baseline criterion that can help you classify your idea, as a feature:
+
+- Requires significant effort/changes to the current BigchainDB implementation and it is not self
+ contained (e.g. API changes,  introducing a new component, takes more than 20 Wrigleys
+ (Wrigley is a `dev day` @ BigchainDB))
+- Impacts the clients and drivers.
+- Can we write a blog post about it?
+- Requires collaboration with multiple teams in terms of development, operations and testing.
+- Impacts documentation significantly.
+
+It is not a feature if:
+
+- It is straightforward enough and might not need a design discussion.
+  - Test case fixes
+  - Fixing typos/grammar in documentation.
+  - Refactoring code
+
+## Classified as a Feature?
+Create a new issue on our [GitHub repository](https://github.com/bigchaindb/bigchaindb/issues/new) with a
+`Feature-Request` label, once you have classified your proposal as a feature and:
+
+- Scratched the surface.
+  - Discussed the proposal in the community e.g. on [our Gitter channel](https://gitter.im/bigchaindb/bigchaindb),
+  Or [Twitter](https://twitter.com/BigchainDB) and saw interest.
+- *Optional:* Have a working prototype.
+- Worked out the resources and people, who can contribute towards this feature.
+
+## Submit Spec
+Submit a design specification document (Pull Request) of your feature. The spec needs to be submitted at:
+
+```text
+/path/to/repo/bigchaindb/docs/specs/features/release-<latest-release>/
```

Comment:
I wonder if we need the path to be so deeply buried. Why not something closer to the top of the repository file tree? The `docs/` directory is mainly for Sphinx source files, but we have lots of README files etc. outside of that already. If you do change it, remember to update the `RELEASE_PROCESS.md` file as well.

Reply:
I think `docs/specs` makes more sense, but you also need release based tracking hence, adding `/features/release-<latest-release>`, I can remove the `/features/` directory and keep it directly under `/path/to/repo/bigchaindb/docs/specs/release-<latest-release>/` ?

Reply:
Yes that would be an improvement.

Reply:
I see that's what you did.